clang/lib/Headers/stdatomic.h

```diff
@@ -104,6 +107,7 @@ typedef _Atomic(long)               atomic_long;
 typedef _Atomic(unsigned long)      atomic_ulong;
 typedef _Atomic(long long)          atomic_llong;
 typedef _Atomic(unsigned long long) atomic_ullong;
+typedef _Atomic(unsigned char)      atomic_char8_t;
 typedef _Atomic(uint_least16_t)     atomic_char16_t;
 typedef _Atomic(uint_least32_t)     atomic_char32_t;
```

Comment:
Should the definition of `atomic_char8_t` also be gated behind C23/C++20?

Also in C++20, this should be `_Atomic(char8_t)`, but the next two are also wrong in C++11 (`_Atomic(char16_t)`/`_Atomic(char32_t)`)

Reply:
Let's address that one in a separate PR, but yes `atomic_char8_t` should be gated behind C23 because the identifier isn't reserved until C23. For C++, it should be gated behind C++20.

As for the definition, technically in C they should be using `char8_t` and friends as well, but those are defined in `uchar.h` which we don't vend with the compiler (should we be vending that one in freestanding?). GCC seems to use `unsigned char` directly rather than use a transitive include, so I think that approach is fine for us to take as well.

Reply:
It seems like this header doesn't properly implement [P0943R6](https://wg21.link/P0943R6) in C++23 (Where `#define _Atomic(T) std::atomic<T>` and all these typedefs should just be `using std::atomic_*`.

So `::atomic_char8_t` should be `std::atomic<char8_t>`, not `std::atomic<unsigned char>` (nor `unsigned char _Atomic`), but that is outside the scope of this PR. I'll just make this match the surrounding typedefs for now.